Sources/Core/FileProtocol.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+/**
+    Objects conforming to this protocol
+    can load and save files to a persistent
+    data store.
+*/
+public protocol FileProtocol {
+    func load(path: String) throws -> Bytes
+    func save(bytes: Bytes, to path: String) throws
+}
+
+/**
+    Conforms NSData to FileProtocol.
+*/
+public final class DataFile: FileProtocol {
+    public init() { }
+
+    public enum Error: Swift.Error {
```

Comment:
Small thing, but can we put nested types above class implementations? My preferred object organization would be like:

```
Outer:
    - nested types
    - properties
    - initializers
    - functionality
```